### DISCUSSION TITLE
Populating a `ForwardingProfile` using a custom schema

### DISCUSSION THREAD

**Discussion by zstadler** [onthegomap/planetiler#1017] [Ideas]
As far as I understand, a [custom schema](https://github.com/onthegomap/planetiler/blob/main/planetiler-custommap/README.md) is a simple way to describe multiple layers that may share common sources. A [`ForwardingProfile`](https://github.com/onthegomap/planetiler/blob/main/planetiler-core/src/main/java/com/onthegomap/planetiler/ForwardingProfile.java) and a set of handlers also describe multiple layers that can share common sources. Currently, these two approaches are mutually exclusive.

From a SW architecture point, would it be possible to develop code that will allow the two approaches to co-exist in the same planetiler run? Is such a run, some layers are defined in a schema file while the complex layers are defined using handlers.

**Comment by msbarry** [comment]
Yes this is a logical next step that would make a lot of sense. Ideally you could compose 2 java profiles in a single run, a java profile and a yaml profile, or multiple yaml profiles so they can share the overhead of a single planetiler run (node location lookups, sort step, etc.) and also output into a single archive. 

To compose raw custom schema files, the layer definitions and tile post-processing compose pretty easily, but the tricky part is what to do with conflicting definitions for schema name, description, attribution, tag mappings, overlay vs. baselayer, and sources. For example you might compose yaml schemas that define the same source with different names/URLs and you want to collapse them down to a single source.

Maybe we need a yaml subset for "composable profile" that only includes the composable stuff, or maybe we just need to introduce a top-level yaml or java layer that defines the non-composable parts (inputs/outputs), points to the yaml/java profiles to be composed and ignores all of the non-composable parts from those files?

Also, I'm assuming the goal is to get the composed profiles into a single archive from a single run of planetiler right? Generating multiple archives from a single planetiler run would theoretically be possible but not sure if there's a use-case to warrants supporting the extra complexity...

**Comment by cmahnke** [comment]
For #1389 I would also like such a feature, my use case would be to (re-) use the current default profile but add some more layers (main use case would be trees and statues). I'm aware that I can use the current approach (as described [here](https://github.com/onthegomap/planetiler#generating-custom-vector-tiles)), so it's not strictly required. But having the possibility to compose a schema from the buildin default and a YAML file would reduce the effort considerably.
